src/ui_basic/panel.h

```diff
@@ -97,6 +97,10 @@ class Panel : public boost::signals2::trackable {
 
 	void free_children();
 
+	bool is_dying() const {
```

Comment:
// Stop think()ing and other activities when we are dying to prevent nasty side effects.